src/Chr.Avro.Codegen/Codegen/CSharpCodeGenerator.cs

```diff
@@ -333,6 +368,32 @@ protected virtual TypeSyntax GetPropertyType(Schema schema, bool nullable = fals
 
                     try
                     {
+                        var recordSchemas = others.OfType<RecordSchema>().ToArray();
+
+                        if (recordSchemas.Length > 1)
+                        {
+                            string? fullName = null;
+
+                            // all record schemas should have been added to the interfaceDeclarationsMap for the same interface
+                            foreach (RecordSchema recordSchema in recordSchemas)
+                            {
+                                if (interfaceDeclarationsMap.ContainsKey(recordSchema.FullName))
+                                {
+                                    fullName = recordSchema.FullName;
+                                    break;
+                                }
+                            }
+
+                            if (!string.IsNullOrWhiteSpace(fullName) &&
```

Comment:
The schema model protects you a bit here, full name is guaranteed to be non-empty and formatted correctly (https://github.com/ch-robinson/dotnet-avro/blob/main/src/Chr.Avro/Abstract/NamedSchema.cs#L82-L95)